dataprofiler/tests/profilers/test_datetime_column_profile.py

```diff
@@ -500,3 +500,26 @@ def test_json_encode_after_update(self):
         )
 
         self.assertEqual(serialized, expected)
+
+    def test_json_decode(self):
+        fake_profile_name = None
+        expected_profile = DateTimeColumn(fake_profile_name)
+
+        serialized = json.dumps(expected_profile, cls=ProfileEncoder)
+        deserialized = load_column_profile(json.loads(serialized))
+
+        test_utils.assert_profiles_equal(deserialized, expected_profile)
+
+    def test_json_decode_after_update(self):
+        fake_profile_name = "Fake profile name"
+
+        data = [2.5, 12.5, "2013-03-10 15:43:30", 5, "03/10/13 15:43", "Mar 11, 2013"]
+        df = pd.Series(data)
+
+        expected_profile = DateTimeColumn(fake_profile_name)
+        expected_profile.update(df)
+
+        serialized = json.dumps(expected_profile, cls=ProfileEncoder)
+        deserialized = load_column_profile(json.loads(serialized))
+
+        test_utils.assert_profiles_equal(deserialized, expected_profile)
```

Comment:
I think we are missing 1 extra test here. That's to validate that after deserialization into a profile again, it can still update. Hence, we need to update with a list of dates that contain a new format and something that might change the min or max. Then validate that. Otherwise, pretty clean!

Reply:
the test currently uses all 3 of the expected formats used in other tests. Do we want to add a 4th format?

Reply:
in the init we have a list of possible formats:
```python
        "%Y-%m-%d %H:%M:%S",  # 2013-03-5 15:43:30
        "%Y-%m-%dT%H:%M:%S",  # 2013-03-6T15:43:30
        "%Y-%m-%dT%H:%M:%S.%fZ",  # 2013-03-6T15:43:30.123456Z
        "%Y-%m-%dt%H:%M:%S.%fz",  # 2013-03-6t15:43:30.123456z
        "%m/%d/%y %H:%M",  # 03/10/13 15:43
        "%m/%d/%Y %H:%M",  # 3/8/2013 15:43
        "%Y%m%dT%H%M%S",  # 2013036T154330
        "%Y-%m-%d",  # 2013-03-7
        "%m/%d/%Y",  # 3/8/2013
        "%m/%d/%y",  # 03/10/13
        "%b %d, %Y",  # Mar 11, 2013
        "%B %d, %Y",  # March 9, 2013
        "%d%b%y",  # 12Mar13
        "%b-%d-%y",  # Mar-13-13
        "%m%d%Y",  # 03142013
        "%H:%M:%S.%f",  # 05:46:30.258509
```
We can select one or two varieties and ensure the updates occur to the list

Reply:
Cool! adding a few now